xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/select/select.js

```diff
@@ -51,11 +59,189 @@ require(['jquery'], function($) {
         return;
       }
       input.prop('checked', true);
-      self.selectWidget.find('.xwiki-select-option-selected').removeClass('xwiki-select-option-selected');
+      // Clear the previous option selected
+      let previousSelection = self.selectWidget.find('.xwiki-select-option-selected');
+      previousSelection.removeClass('xwiki-select-option-selected');
+      previousSelection.removeAttr('aria-selected');
+      // Clear the previous option highlighted
+      let previousHighlight = self.selectWidget.find('.xwiki-select-option-highlighted');
+      if (previousHighlight) previousHighlight.removeClass('xwiki-select-option-highlighted');
+      // Set aria-activedescendant on the second level parent, which should be the listbox object.
+      option.parent().parent().attr('aria-activedescendant', option.attr('id'));
+      // Update the new highlighted option
+      self.changeHighlight(option);
+      // Update the new selected option
       option.addClass('xwiki-select-option-selected');
+      option.attr('aria-selected', 'true');
       self.triggerSelectionChange();
     };
 
+    /**
+     * Highlights the option given as a parameter.
+     * @since 15.5RC1
```

Comment:
The since-versions here and below need to be updated.

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2182/commits/00b30139f8961bb7646c0f6595d6d25eda2a0d69